mlrun/feature_store/retrieval/spark_merger.py

```diff
@@ -240,8 +163,23 @@ def _join(
                 be prefixed with featureset_df name.
 
         """
-        indexes = list(featureset.spec.entities.keys())
-        merged_df = entity_df.join(featureset_df, on=indexes)
+        # fs_name = featureset.metadata.name
+        join_cond = None
+        if left_keys != right_keys:
+            for key_l, key_r in zip(left_keys, right_keys):
+                join_cond = (
+                    join_cond & (entity_df[key_l] == featureset_df[key_r])
+                    if join_cond
+                    else (entity_df[key_l] == featureset_df[key_r])
+                )
```

Comment:
```suggestion
join_cond = [entity_df[key_l] == featureset_df[key_r] for key_l, key_r in zip(left_keys, right_keys)]
```
Spark accepts an array of conditions and `and`s them.